app/imports/components/posts/NewPost.jsx

```diff
@@ -16,30 +16,74 @@ export class NewPost extends Component {
     this.state = {
       photo,
       fingerprint,
-      caption: ''
+      caption: '',
+      uploading: false,
     };
 
     this.onInputChange = this.onInputChange.bind(this);
     this.onSubmitForm = this.onSubmitForm.bind(this);
+    this.handleUpload = this.handleUpload.bind(this);
   }
 
-  onInputChange(event) {
-    this.setState({caption: event.target.value});
+  getSlingshowUploader() {
+    return new Slingshot.Upload('imageUpload');
   }
 
-  onSubmitForm(event) {
-    event.preventDefault();
-
+  uploadFile() {
+    const uploader = this.getSlingshowUploader();
     const photo = this.state.photo;
+
+    this.setState({
+      uploading: true,
+    });
+
+    // Use fetch to convert base64 to blob
+    fetch(photo)
+      .then( res => res.blob() )
+      .then( blob => {
+        uploader.send(blob, this.handleUpload);
+      });
+  }
+
+  handleUpload(error, url) {
     const caption = this.state.caption;
     const fingerprint = this.state.fingerprint;
 
-    addPost.call({
-      photo,
-      caption,
-      fingerprint,
+    this.setState({
```

Comment:
perhaps this should only reset on fail as we don't want any more uploads if successful